packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -339,6 +339,19 @@ class FlutterPlugin implements Plugin<Project> {
                 implementation("androidx.multidex:multidex:2.0.1")
             }
             }
+
+        // Validate that the provided Gradle, Java, AGP, and KGP versions are all within our
+        // supported range.
+        if (!project.hasProperty("skipDependencyChecks") || !project.getProperty("skipDependencyChecks")) {
+            try {
+                project.apply from: Paths.get(flutterRoot.absolutePath, "packages", "flutter_tools",
+                        "gradle", "src", "main", "kotlin", "dependency_version_checker.gradle.kts")
+            } catch (Exception ignored) {
```

Comment:
Is there a more specific exception you are expecting? 
Consider logging the ignored exception information.

Reply:
Added a log. I don't have a more specific exception. I just don't want this to block a build under any circumstances and am slightly apprehensive because of the small amount of reflection used in getting the AGP and KGP versions. I can remove if you prefer

Reply:
Nope that explanation makes sense. If you were specifically worried about file permissions for example then catching that exception would be better than a general catch all.